app/scripts/views/PlacesEdit.js

```diff
@@ -52,27 +55,34 @@ class PlacesEdit extends Component {
     const { title, description, isPublic, images } = values
     const { slots } = values.slots
     const disabledSlots = slots ? getDisabledSlotIndexes(slots) : []
-
+    const th = this
     const successFlash = {
       text: translate('flash.updatePlaceSuccess'),
     }
-
-    const requestParams = {
-      ...values.location,
-      description,
-      disabledSlots,
-      images,
-      isPublic,
-      title,
-    }
-
-    this.props.updateResource(
-      'places',
-      this.props.resourceSlug,
-      requestParams,
-      successFlash,
-      '/places'
-    )
+    const address = String(values.location.street) + ', ' + String(values.location.cityCode) + ', ' + String(values.location.city) + ', ' + String(values.location.country)
+    provider
+      .search({ query: address })
+      .then(function(result) {
```

Comment:
Check out the new ES6 arrow functions in JavaScript: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

```
.then(result => {
 ...
})
```

They allow you to stay in the binding scope, which enables you to just use `this` without needing to store it in an additional variable before.

Reply:
Thanks! That'll teach me to copy and paste all the time ;-)